packages/cssnano-util-same-parent/src/index.js

```diff
@@ -6,9 +14,11 @@ export default function sameParent (ruleA, ruleB) {
     }
     // If an at rule, ensure that the parameters are the same
     if (ruleA.parent.type === 'atrule' && ruleB.parent.type === 'atrule') {
+        const ruleATopAtRuleParent = getTopAtRuleParent(ruleA.parent);
+        const ruleBTopAtRuleParent = getTopAtRuleParent(ruleB.parent);
```

Comment:
@andyjansson why? please provide example

Reply:
Maybe we can add `top` (by default `false`) argument to `sameParent` function and use this when necessary

Reply:
This util function was made to determine whether two elements shares the same parent. This is not what your code changes have done. You've changed it so that it checks whether the top ancestor is the same, which is not the same as the parent. The function no longer does what it claims it does. 

Additionally, this looks like it could lead to some nasty bugs when dealing with at-rules nested in parallel.

Reply:
With this change, if the parent is an at-rule, then it ignores the parent's at-rule params, and only checks its parent's parent's at-rules params. This would be a match:

````
@media (width: 40px) {
  @supports (pointer: fine) {
    .a { color: red; }
}

@media (width: 40px) {
  @supports (pointer: course) {
    .a { background-color: blue; }
}
````

Since this commit is rather old (almost 6 months old), I'm working on making pull request to fix this issue, and perhaps creating a new mergeAtRules package.